var/spack/repos/builtin/packages/legion/package.py

```diff
@@ -46,10 +47,11 @@ class Legion(CMakePackage, ROCmPackage):
     # use for general (not legion development) use cases.
     depends_on("mpi", when="network=mpi")
     depends_on("mpi", when="network=gasnet")  # MPI is required to build gasnet (needs mpicc).
+    depends_on("ucx", when="network=ucx")
     depends_on("ucx", when="conduit=ucx")
     depends_on("mpi", when="conduit=mpi")
-    depends_on("cuda@10.0:11.9", when="+cuda_unsupported_compiler")
-    depends_on("cuda@10.0:11.9", when="+cuda")
+    depends_on("cuda@10.0:12.2", when="+cuda_unsupported_compiler")
+    depends_on("cuda@10.0:12.2", when="+cuda")
```

Comment:
@elliottslaughter do all versions of legion support up to `cuda@12.2` or should we be setting these bounds on a per version basis? (Will `legion@21.03.0` compile with `cuda@12.2`?)

Reply:
My understanding is that the existing Legion releases should work with CUDA 12.2, but only Legion 23.06.0 will work with 12.1 (due to a CUDA bug in 12.1). I'm not sure it's worth the effort to track version bounds at that granularity? Aside from CUDA bugs like this Legion usually works without any additional effort.

Reply:
An alternative would be just to say that we'll only add new CUDA versions for new Legion versions. That will be much more conservative (because usually those releases would work fine), but I'm not really interested in debugging compatibility issues on older Legions anyway, so it seems like the safer (and possibly saner) approach.

Reply:
@alecbcs Ok, I attempted to resolve this in the most recent version (taking tge more conservative approach in my second comment). Could you check to see if this looks reasonable?